app/models/alchemy/element.rb

```diff
@@ -300,23 +313,23 @@ def copy_nested_elements_to(target_element)
       nested_elements.map do |nested_element|
         Element.copy(nested_element, {
           parent_element_id: target_element.id,
-          page_id: target_element.page_id,
+          page_version_id: target_element.page_version_id,
           cell_id: target_element.cell_id
         })
       end
     end
 
     private
 
-    # Returns previous or next public element from same page.
+    # Returns previous or next public element from same page version.
     #
     # @param [String]
     #   Pass '>' or '<' to find next or previous public element.
     # @param [String]
     #   Pass an element name to get previous of this kind.
     #
     def previous_or_next(dir, name = nil)
-      elements = page.elements.published.where("#{self.class.table_name}.position #{dir} #{position}")
+      elements = siblings.published.where("#{Element.table_name}.position #{dir} #{position}")
```

Comment:
As I don't think this is a severe security issue, if any at all, it's ok to send a public PR IMO. Thanks for asking.